libqfieldsync/project_checker.py

```diff
@@ -439,3 +444,47 @@ def check_external_layers(
             )
 
         return None
+
+    def check_layer_has_experimental_cloud_support(
+        self, layer_source: LayerSource
+    ) -> Optional[FeedbackResult]:
+        """Check if layer has experimental cloud support"""
+        layer = layer_source.layer
+
+        # Check if layer is marked for offline editing
+        if layer_source.cloud_action != SyncAction.OFFLINE:
+            return None
+
+        # Check if layer is read-only
+        if layer.readOnly():
+            return None
+
+        # Check if layer is in a supported format
+        provider_type = layer.providerType()
+
+        # Check if it's a GeoPackage
+        if layer.dataProvider().storageType() == "GPKG":
+            return None
```

Comment:
```suggestion
        # We support GeoPackages on the cloud, so return early
        if storage_type == "GPKG":
            return None
```

If we want to keep these comments, we better add a bit more value to them.

Also switched to the new variable I have introduced. Saves a few bytes. This time it is acceptable "renaming" of a variable, makes code shorter, easier to read and saves a few CPU cycles.